constants/limit.go

```diff
@@ -17,4 +17,7 @@ const (
 
 	// FavoritesMaxSize defines the maximum size in characters for user favorites.
 	FavoritesMaxSize = 5000
+
+	// RepoNameHistoryMax defines the maximum size in characters for repo name history.
+	RepoNameHistoryMax = 5000
```

Comment:
Could you elaborate on why we chose `5000` here?

I'm wondering if this value should be smaller since it seems a bit large.

Based off the PR description, it doesn't sound like we have a use-case for needing a long history.

This has me leaning towards a `n-1` approach for the number of old repo names we keep on record.

Reply:
Do you think we would be better served just having the `NameHistory` be just `PreviousName` and have it just be a string type? Perhaps a character limit of 100 or so might be a better idea if we wanted to keep NameHistory. Also, thinking more about this, we would need some sort of clear space logic (rather than rejection logic seen in Favorites) if we didn't want to run into any issues down the line.

Reply:
I decided to go ahead and give the slice a limit of 100. I also added some truncation logic in the setter method in order to keep an up-to-date name history without erroring out or running over the max. Let me know what you think!

Reply:
> Do you think we would be better served just having the NameHistory be PreviousName and have it be a string type?

That would be my preference since we don't have a use case for needing a longer history.

If I'm following correctly, any names older than `n-1` would be considered stale.

If there is a need to have a longer history than that, I would rather defer to the SCM systems audit/history.

All that being said, perhaps we should wait to hear feedback/thoughts from others?

Unless we want me to be the sole individual responsible for making this decision 😅 

> ... we would need some sort of clear space logic (rather than rejection logic seen in Favorites) when the limit is reached if we didn't want to run into any issues down the line.

This debatably could be another reason to only preserve the last name for a repo rather than multiple.

I could see a scenario for confusion if one member of a team renames a repo and a different member was unaware of this change and now they're confused what happened to their repo and why it is no longer in their favorites.

Reply:
The only thing I can think of is if we wanted to do some database clean up with repos that had no related SCM instance, we could use the NameHistory to find those.  Other than that though, I cannot see a possible reason for holding on to them.

Reply:
If I'm following correctly, I think that use case could still be accomplished with `n-1` repo name history 👍 

It sounds like the idea would be to remove any DB records from the `repos` table that don't match:

* the `repo.Name` field for any existing repo
* anything inside the `repo.NameHistory` field

If so, then I think this would also work for checking `repo.PreviousName` instead of checking `repo.NameHistory`.

Reply:
I apologize -- I misinterpreted my own code. Since this does an `update`, the `NameHistory` field wouldn't have any connection to repos that are in the database. So I think having `repo.PreviousName` be a string value is the way to go.

Reply:
No worries 😃 

It's all good 👍 

FWIW:

You could potentially wait to see if others have thoughts/feedback on the discussion before making a change.

Reply:
i like the thought here 👍🏼 

I see the length was changed to 100. i think GitHub's limit is 100. However, Gitlab has a limit of 255 if i'm not mistaken. Wonder if we should take inventory quick of max length from, at least, major SCM providers to see what makes the most sense there.

Reply:
I think `RepoNameHistoryMax` could be removed if we decide to only store one previous repo name as a `string` 👍

Reply:
I just pushed up what the code would look like if we went with just a `string` type.  Can easily be reverted back if we decide to do a slice.